File: plsql-squid/src/test/java/br/com/felipezorzo/sonar/plsql/api/RefDatatypeTest.java
package br.com.felipezorzo.sonar.plsql.api;

import static org.sonar.sslr.tests.Assertions.assertThat;

import org.junit.Before;
import org.junit.Test;

public class RefDatatypeTest extends RuleTest {

    @Before
    public void init() {
        setRootRule(PlSqlGrammar.REF_DATATYPE);
    }
    
    @Test
    public void matchesSimpleRef() {
        assertThat(p).matches("ref custom");
    }
    
    @Test
    public void matchesRefCursor() {
        assertThat(p).matches("ref cursor");
    }

}

File: zpa-core/src/test/kotlin/org/sonar/plugins/plsqlopen/api/RefDatatypeTest.kt
/**
 * Z PL/SQL Analyzer
 * Copyright (C) 2015-2024 Felipe Zorzo
 * mailto:felipe AT felipezorzo DOT com DOT br
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package org.sonar.plugins.plsqlopen.api

import com.felipebz.flr.tests.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class RefDatatypeTest : RuleTest() {

    @BeforeEach
    fun init() {
        setRootRule(PlSqlGrammar.REF_DATATYPE)
    }

    @Test
    fun matchesSimpleRef() {
        assertThat(p).matches("ref custom")
    }

}
